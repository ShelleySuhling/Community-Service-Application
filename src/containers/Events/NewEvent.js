import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as eventsActions from '../../actions/eventsActions';
import PropTypes from 'prop-types';
import EventForm from '../../components/Forms/EventForm'
import { Redirect } from 'react-router-dom'

class NewEvent extends Component {

    constructor(props) {
        super(props)
        this.state = {
            event: {}
        }
    }

    onSubmit = (event) => {
        this.props.eventsActions.requestCreateEvent(event)
    }

    render() {
        let { events } = this.props
        if (events.newEvent) {
            return <Redirect to='/events' />
        } else {
            return (
                <div className="content-container">
                    <EventForm
                        onSubmit={this.onSubmit}
                        submitText="Create Event" />
                </div >)
        }
    }
}

NewEvent.propTypes = {
    eventsActions: PropTypes.object,
    events: PropTypes.object
};
function mapStateToProps(state) {
    return {
        events: state.events,
    };
}
function mapDispatchToProps(dispatch) {
    return {
        eventsActions: bindActionCreators(eventsActions, dispatch)
    };
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(NewEvent);