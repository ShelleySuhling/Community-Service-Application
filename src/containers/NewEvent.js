import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as eventsActions from '../actions/eventsActions';
import PropTypes from 'prop-types';
import EventForm from '../components/EventForm'
import { Redirect } from 'react-router-dom'
import * as _ from 'lodash'
import moment from 'moment'
import FormErrors from '../components/FormErrors'



class NewEvent extends Component {

    constructor(props) {
        super(props)
        this.state = {
            title: '',
            location: '',
            startTime: null,
            endTime: null,
            date: null,
            isMandatory: false,
            isValid: false,
            errors: []
        }
    }

    handleChange = (name, value) => {
        this.setState({
            ...this.state,
            [name]: value
        });
    }

    validateState = () => {
        let { title, location, startTime, endTime, date, isMandatory } = this.state
        let errors = []
        if (!title || title.length <= 5) {
            errors.push("Title must be longer than 5 characters.")
        }
        if (!location || location.length <= 5) {
            errors.push("Location must be longer than 5 characters.")
        }
        if (!startTime || moment(date).hours(startTime).isBefore(moment())) {
            errors.push("Start must be in future")
        }
        if (!endTime || moment(date).hours(endTime).isBefore(moment())) {
            errors.push("End must be in future")
        }
        if (!date || moment(date).isBefore(moment())) {
            errors.push("Date must be in future")
        }

        this.setState({
            ...this.state,
            errors: errors
        })
        return errors

    }

    toggleMandatory = () => {
        this.setState({
            isMandatory: !this.state.isMandatory
        })
    }

    onSubmit = () => {
        console.log('****', this.validateState())
        if (this.validateState() == 0) {
            this.props.eventsActions.requestCreateEvent(this.state)
        } else {
            window.scrollTo(0, 0)
        }
    }

    render() {
        let { events } = this.props
        console.log(this.state)
        if (events.newEvent) {
            return <Redirect to='/events' />
        } else {
            return (
                <div className="content-container">
                    <FormErrors errors={this.state.errors} />
                    <EventForm event={this.state}
                        handleChange={this.handleChange}
                        toggleMandatory={this.toggleMandatory}
                        onSubmit={this.onSubmit}
                        submitText="Create Event" />
                </div >)
        }
    }
}

NewEvent.propTypes = {
    eventsActions: PropTypes.object,
    events: PropTypes.object
};
function mapStateToProps(state) {
    return {
        events: state.events,
    };
}
function mapDispatchToProps(dispatch) {
    return {
        eventsActions: bindActionCreators(eventsActions, dispatch)
    };
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(NewEvent);