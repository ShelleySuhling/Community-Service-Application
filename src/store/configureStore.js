import { applyMiddleware, createStore } from 'redux'
import { persistStore, persistCombineReducers } from 'redux-persist';
import thunkMiddleware from 'redux-thunk'
import storage from 'redux-persist/lib/storage';

import rootReducer from '../reducers/rootReducer'

export default function configureStore(preloadedState) {

    const persistConfig = {
        key: 'root',
        storage: storage,
    };
    const pReducer = persistCombineReducers(persistConfig, rootReducer);

    const store = createStore(pReducer, applyMiddleware(thunkMiddleware));
    const persistor = persistStore(store);
    window.store = store;

    return { persistor, store }
}