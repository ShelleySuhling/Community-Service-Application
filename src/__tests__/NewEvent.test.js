import * as React from 'react';
import { shallow } from 'enzyme';

import NewEvent from '../containers/Events/NewEvent';

import configureStore from '../store/configureStore';
const { store } = configureStore()

describe('NewEvent', () => {
    const newEventWrapper = shallow(<NewEvent store={store} />).dive()
    it('matches snapshot', () => {
        expect(newEventWrapper).toMatchSnapshot()
    })
    it('has an empty event object in state', () => {
        expect(newEventWrapper.state('event')).toEqual({})
    })
    it('onSubmit calls requestCreateEvent action', () => {
        const request_create_event_mock = jest.fn()
        newEventWrapper.instance().props.eventsActions.requestCreateEvent = request_create_event_mock

        newEventWrapper.instance().onSubmit()
        expect(request_create_event_mock).toHaveBeenCalled()
    })
    it('redirects to events page when event is made', () => {
        const redirect_to_events_mock = jest.fn()
        newEventWrapper.instance().redirectToEvents = redirect_to_events_mock
        newEventWrapper.instance().props.events.newEvent = true

        newEventWrapper.instance().forceUpdate()
        expect(redirect_to_events_mock).toHaveBeenCalled()
    })
})