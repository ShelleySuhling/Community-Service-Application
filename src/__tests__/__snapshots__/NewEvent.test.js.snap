// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewEvent matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewEvent
    events={Object {}}
    eventsActions={
      Object {
        "requestCreateEvent": [Function],
        "requestEditEvent": [Function],
        "requestEvents": [Function],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <EventForm
        onSubmit={[Function]}
        submitText="Create Event"
      />,
      "className": "content-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onSubmit": [Function],
        "submitText": "Create Event",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <EventForm
          onSubmit={[Function]}
          submitText="Create Event"
        />,
        "className": "content-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSubmit": [Function],
          "submitText": "Create Event",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
